src/transformers/models/textnet/convert_textnet_original_to_pytorch.py

```diff
@@ -0,0 +1,202 @@
+    namespace = {}
+
+    exec(content, namespace)
+
+    model_config = namespace.get("model")
+    data_config = namespace.get("data")
+    size = 640
+    if "train" in data_config:
+        if "short_size" in data_config["train"]:
+            size = data_config["train"]["short_size"]
+
+    if "tiny" in model_config["backbone"]["config"]:
+        config = prepare_config(tiny_config_url, size)
+    elif "small" in model_config["backbone"]["config"]:
+        config = prepare_config(small_config_url, size)
+    else:
+        config = prepare_config(base_config_url, size)
+
+    model = TextNetBackbone(config)
+    textnet_image_processor = CLIPImageProcessor(
+        size={"shortest_edge": size}, do_center_crop=False, use_square_size=True
+    )
+    state_dict = torch.hub.load_state_dict_from_url(checkpoint_url, map_location="cpu", check_hash=True)["ema"]
+    state_dict_changed = OrderedDict()
+    for key in state_dict:
+        if "backbone" in key:
+            val = state_dict[key]
+            new_key = key
+            for search, replacement in rename_key_mappings.items():
+                if search in new_key:
+                    new_key = new_key.replace(search, replacement)
+
+            pattern = r"textnet\.stage(\d)"
+
+            def adjust_stage(match):
+                stage_number = int(match.group(1)) - 1
+                return f"textnet.encoder.stages.{stage_number}.stage"
+
+            # Using regex to find and replace the pattern in the string
+            new_key = re.sub(pattern, adjust_stage, new_key)
+            state_dict_changed[new_key] = val
+    model.load_state_dict(state_dict_changed)
+
+    model.save_pretrained(pytorch_dump_folder_path)
+    textnet_image_processor.save_pretrained(pytorch_dump_folder_path)
+    logging.info("The converted weights are save here : " + pytorch_dump_folder_path)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        "--checkpoint_url",
+        default="https://conversationhub.blob.core.windows.net/beit-share-public/beit/beit_base_patch16_224_pt22k_ft22kto1k.pth",
+        type=str,
+        help="URL to the original PyTorch checkpoint (.pth file).",
+    )
+    parser.add_argument(
+        "--checkpoint_config_url",
+        default="https://conversationhub.blob.core.windows.net/beit-share-public/beit/beit_base_patch16_224_pt22k_ft22kto1k.pth",
+        type=str,
+        help="URL to the original PyTorch checkpoint (.pth file).",
+    )
+    parser.add_argument(
+        "--pytorch_dump_folder_path", default=None, type=str, help="Path to the folder to output PyTorch model."
+    )
+    parser.add_argument(
+        "--validate_logits",
+        default=False,
+        type=bool,
+        help="whether to assert logits outputs",
+    )
+    parser.add_argument(
+        "--save_backbone_separately",
+        default=False,
+        type=bool,
+        help="whether to assert logits outputs",
```

Comment:
this needs to be updated

Reply:
I am removing this argument as it is not relavant for textnet.